src/util/expr_initializer.cpp

```diff
@@ -172,21 +172,14 @@ exprt expr_initializert<nondet>::expr_initializer_rec(
     if(tmpval.is_nil())
       return nil_exprt();
 
-    mp_integer vector_size;
-
-    if(to_integer(vector_type.size(), vector_size))
-    {
-      if(nondet)
-        return side_effect_expr_nondett(type, source_location);
-      else
-        return nil_exprt();
-    }
+    const mp_integer vector_size =
+      numeric_cast_v<mp_integer>(vector_type.size());
 
```

Comment:
This is a change in behavior -- intended?

Reply:
Yes, my apologies for not noting that in the commit message (which I have now done). In several places we had already treated vectors having constant size as an invariant. This should add consistency.